tests/Integration/Behaviour/Features/Context/Domain/TaxRulesGroupFeatureContext.php

```diff
@@ -79,4 +92,185 @@ public function assertLastErrorIsTaxRulesGroupNotFound(): void
+        }
+        if (isset($data['is_enabled'])) {
+            $command->setEnabled(PrimitiveUtils::castStringBooleanIntoBoolean($data['is_enabled']));
+        }
+        $this->getCommandBus()->handle($command);
+    }
+
+    /**
+     * @When /^I (enable|disable)? tax rules group "(.*)"$/
+     */
+    public function toggleStatusTaxRulesGroup(bool $expectedStatus, string $taxRulesGroupReference): void
+    {
+        $taxRulesGroupId = SharedStorage::getStorage()->get($taxRulesGroupReference);
+
+        $this->getCommandBus()->handle(new SetTaxRulesGroupStatusCommand($taxRulesGroupId, $expectedStatus));
+    }
+
+    /**
+     * @When I delete tax rules group :taxRulesGroupReference
+     */
+    public function deleteTaxRulesGroup(string $taxRulesGroupReference): void
+    {
+        $taxRulesGroupId = SharedStorage::getStorage()->get($taxRulesGroupReference);
+
+        $this->getCommandBus()->handle(new DeleteTaxRulesGroupCommand($taxRulesGroupId));
+    }
+
+    /**
+     * @When /^I (enable|disable)? tax rules groups: "([^"]*)"$/
+     */
+    public function bulkToggleStatusTaxRulesGroup(bool $expectedStatus, string $taxRulesGroupReference): void
+    {
+        $taxRulesGroupReferences = PrimitiveUtils::castStringArrayIntoArray($taxRulesGroupReference);
+
+        $idsByReference = [];
+        foreach ($taxRulesGroupReferences as $reference) {
+            $idsByReference[$reference] = SharedStorage::getStorage()->get($reference);
+        }
+
+        $this->getCommandBus()->handle(new BulkSetTaxRulesGroupStatusCommand(
+            $idsByReference,
+            $expectedStatus
+        ));
+    }
+
+    /**
+     * @When /^I bulk delete tax rules groups: "([^"]*)"$/
+     */
+    public function bulkDeleteTaxRulesGroup(string $taxRulesGroupReference): void
+    {
+        $taxRulesGroupReferences = PrimitiveUtils::castStringArrayIntoArray($taxRulesGroupReference);
+
+        $idsByReference = [];
+        foreach ($taxRulesGroupReferences as $reference) {
+            $idsByReference[$reference] = SharedStorage::getStorage()->get($reference);
+        }
+
+        $this->getCommandBus()->handle(new BulkDeleteTaxRulesGroupCommand($idsByReference));
+    }
+
+    /**
+     * @Then tax rules group :taxRulesGroupReference name should be :name
+     */
+    public function assertTaxRulesGroupName(string $taxRulesGroupReference, string $name): void
+    {
+        $editableTaxRulesGroup = $this->getEditableTaxRulesGroup($taxRulesGroupReference);
+
+        if ($editableTaxRulesGroup->getName() !== $name) {
+            throw new RuntimeException(sprintf(
+                'Tax Rules Group "%s" has "%s" name, but "%s" was expected.',
+                $taxRulesGroupReference,
+                $editableTaxRulesGroup->getName(),
+                $name
+            ));
+        }
+    }
+
+    /**
+     * @Then /^tax rules group "(.*)" should be (enabled|disabled)?$/
+     * @Given /^tax rules group "(.*)" is (enabled|disabled)?$/
```

Comment:
Why the `?`? Am I missing something?
```suggestion
     * @Then /^tax rules group "(.*)" should be (enabled|disabled)$/
     * @Given /^tax rules group "(.*)" is (enabled|disabled)$/
```

Reply:
Fixed